8.x/theme-development/client-side-cart-module.md

```diff
@@ -0,0 +1,665 @@
+        const thisEl = jQuery(this);
+    const pId = thisEl.data('product-id');
+    const discountContainerEl = thisEl.find(".discount-container");
+    let priceWithoutDiscount = parseInt(thisEl.attr('product-price'));
+    let count = jQuery('[data-product-id=' + pId + '] .counter-box-' + pId + ' .count-control').val();
+    count = tools.convertNumberToEnglish(count);
+
+     const discountGroup = thisEl.data("discount-group");
+    let isDiscountPercentage = false;
+    let discount = 0;
+    const specialDiscount = discountContainerEl.find(".price-data.discount-value.special");
+    if (specialDiscount.length > 0) {
+        discount = specialDiscount.data("amount");
+    } else if (discountGroup !== null && typeof discountGroup !== "undefined") {
+         isDiscountPercentage = discountGroup.is_percentage;
+        discount = tools.calculateDiscount(discountGroup, count, priceWithoutDiscount);
+    if (discountContainerEl.hasClass("discount-list")) {
+                            discountContainerEl.find("li.active").removeClass("active");
+        discountContainerEl.find(`li[data-discount-value='${discount}']`).addClass("active");
+    } else {
+        discountContainerEl.find(".discount-value").text(tools.convertNumberToPersian(`${discount}`));
+     }
+}
+
+```
+The next step calculates **SumProductPrice** for each product and updates the value. If **count>0**, it calculates the discount according to the number of products, and the value is updated. Otherwise, the discount is applied for one product.
+ 
+```js
+
+   loadedIds.push(`${pId}`);
+         if (count > 0) {
+            let priceAfterDiscount = count * (priceWithoutDiscount - (isDiscountPercentage(priceWithoutDiscount * discount / 100) : discount));
+            priceWithoutDiscount = count * priceWithoutDiscount;
+            finalPriceBeforeDiscount += priceWithoutDiscount;
+            finalPriceAfterDiscount += priceAfterDiscount;
+            LocalCartService.updateSumProductPrice(pId, priceAfterDiscount, priceWithoutDiscount);
+        } else {
+            let priceAfterDiscount = priceWithoutDiscount - (isDiscountPercentage(priceWithoutDiscount * discount / 100) : discount);
+            LocalCartService.updateSumProductPrice(pId, priceAfterDiscount, priceWithoutDiscount);
+         }
+
+     });
+```
+
+After calculating the final price before and after the discount of each product, the total cost of the products in the shopping cart is calculated and displayed at the bottom of the product page.
+
+![directCalculateInvoice.png](/directCalculateInvoice.png)
+
+The cartData object stores the data of the products in the shopping cart. This section returns products that are not on the product page. For example, the cartData keys of a shopping cart are the following array {673,676,124}, the products that are not on the product page are filtered, and this array {673 and 676} is returned. Finally, foreach is performed on these two IDs, and the data inside is calculated.
```

Comment:
Note: All over the document, please note that it is necessary to put the variable names, pieces of codes, and any other programmatically content should be placed in between two grave signs.
For example:
 - cartData -> `cartData`,
 - {673, 676, 124} -> `{673, 676, 124}`
this makes your document more professional.

Reply:
Should I put the names of the functions in the grave signs?
But I bold the variables and functions and placed the URLs in the grave signs.
Like these:
so the **setExtraFeeAmount** function takes a new amount from the input and makes the **ExtraFeeAmount** variable equal to that new amount, and finally, the new invoice calculates.


For example, the **cartData** keys of a shopping cart are the following array `{673,676,124}`

Reply:
I actually prefer the grave sign for programmatic parts. Please consider changing them.